Spigot-Server-Patches/0182-Update-VehicleCreateEvent.patch

```diff
@@ -0,0 +1,136 @@
+     }
+ 
+     protected boolean playStepSound() {
+diff --git a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
+index 9df7dfc..a4a2c19 100644
+--- a/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
++++ b/src/main/java/net/minecraft/server/EntityMinecartAbstract.java
+@@ -111,7 +111,7 @@ public abstract class EntityMinecartAbstract extends Entity implements INamableT
+         this.lastY = d1;
+         this.lastZ = d2;
+ 
+-        this.world.getServer().getPluginManager().callEvent(new org.bukkit.event.vehicle.VehicleCreateEvent((Vehicle) this.getBukkitEntity())); // CraftBukkit
++        //this.world.getServer().getPluginManager().callEvent(new org.bukkit.event.vehicle.VehicleCreateEvent((Vehicle) this.getBukkitEntity())); // CraftBukkit // Paper - Relocated elsewhere
+     }
+ 
+     public double ay() {
+diff --git a/src/main/java/net/minecraft/server/ItemBoat.java b/src/main/java/net/minecraft/server/ItemBoat.java
+index c68e328..86f080e 100644
+--- a/src/main/java/net/minecraft/server/ItemBoat.java
++++ b/src/main/java/net/minecraft/server/ItemBoat.java
+@@ -66,6 +66,15 @@ public class ItemBoat extends Item {
+                 boolean flag1 = block == Blocks.WATER || block == Blocks.FLOWING_WATER;
+                 EntityBoat entityboat = new EntityBoat(world, movingobjectposition.pos.x, flag1 ? movingobjectposition.pos.y - 0.12D : movingobjectposition.pos.y, movingobjectposition.pos.z);
+ 
++                // Paper start
++                EntityPlayer entityplayer = (EntityPlayer) entityhuman;
++
++                if (org.bukkit.craftbukkit.event.CraftEventFactory.callVehicleCreateEvent(entityboat, entityplayer.getBukkitEntity(), org.bukkit.event.vehicle.VehicleCreateEvent.VehicleCreateReason.PLAYER).isCancelled()) {
++                    entityplayer.updateInventory(entityplayer.activeContainer); // Update inventory as the client will think there is no item in hand otherwise
++                    return new InteractionResultWrapper(EnumInteractionResult.PASS, itemstack);
++                }
++                // Paper end
++
+                 entityboat.setType(this.a);
+                 entityboat.yaw = entityhuman.yaw;
+                 if (!world.getCubes(entityboat, entityboat.getBoundingBox().g(-0.1D)).isEmpty()) {
+diff --git a/src/main/java/net/minecraft/server/ItemMinecart.java b/src/main/java/net/minecraft/server/ItemMinecart.java
+index 46c3e76..4c5d05f 100644
+--- a/src/main/java/net/minecraft/server/ItemMinecart.java
++++ b/src/main/java/net/minecraft/server/ItemMinecart.java
+@@ -5,6 +5,11 @@ import org.bukkit.craftbukkit.inventory.CraftItemStack;
+ import org.bukkit.event.block.BlockDispenseEvent;
+ // CraftBukkit end
+ 
++// Paper start
++import org.bukkit.craftbukkit.event.CraftEventFactory;
```

Comment:
I was using the craft event factory in two places, so i still need to fully qualify them?